sdk/core/azure-core/azure/core/rest/_aiohttp.py

```diff
@@ -0,0 +1,147 @@
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+
+import asyncio
+import codecs
+from typing import AsyncIterator
+from multidict import CIMultiDict
+import aiohttp
+from . import HttpRequest, AsyncHttpResponse
+try:
+    import cchardet as chardet
+except ImportError:  # pragma: no cover
+    import chardet  # type: ignore
+from ._helpers_py3 import iter_raw_helper, iter_bytes_helper
+from ..pipeline.transport._aiohttp import AioHttpStreamDownloadGenerator
+from ..exceptions import ResponseNotReadError
+
+
+class RestAioHttpTransportResponse(AsyncHttpResponse):
+    def __init__(
+        self,
+        *,
+        request: HttpRequest,
+        internal_response,
+    ):
+        super().__init__(request=request, internal_response=internal_response)
+        self.status_code = internal_response.status
+        self.headers = CIMultiDict(internal_response.headers)  # type: ignore
+        self.reason = internal_response.reason
+        self.content_type = internal_response.headers.get('content-type')
+        self._decompress = None
+        self._decompressed_content = None
+
+    @property
+    def content(self) -> bytes:
+        if self._content is None:
+            raise ResponseNotReadError()
+        if not self._decompress:
+            return self._content
+        enc = self.headers.get('Content-Encoding')
+        if not enc:
+            return self._content
+        enc = enc.lower()
+        if enc in ("gzip", "deflate"):
+            if self._decompressed_content:
+                return self._decompressed_content
+            import zlib
+            zlib_mode = 16 + zlib.MAX_WBITS if enc == "gzip" else zlib.MAX_WBITS
+            decompressor = zlib.decompressobj(wbits=zlib_mode)
+            self._decompressed_content = decompressor.decompress(self._content)
+            return self._decompressed_content
+        return self._content
+
+    @property
+    def text(self) -> str:
+        content = self.content
+        encoding = self.encoding
+        ctype = self.headers.get(aiohttp.hdrs.CONTENT_TYPE, "").lower()
+        mimetype = aiohttp.helpers.parse_mimetype(ctype)
+
+        encoding = mimetype.parameters.get("charset")
```

Comment:
Given we set encoding here, why we need L80?

Reply:
I'm following the strucutre for the code in pipeline transport. But basically, it seems that if the chardet detecting fails, encoding still might be none, so we fall back to "utf-8-sig"

Reply:
I mean L84 will override L80, right?

Reply:
I cleaned this up a bit, lmk your thoughts, thanks!